### PR TITLE
Fix: warning pedantic range expr

### PR DESCRIPTION
**Pull request by perigoso**
## Detailed description

This one was missed from the initial batch

Fix (some) warnings for `-Wpedantic`

See #1590 for context

## Your checklist for this pull request

* [x] I've read the [Code of Conduct](https://github.com/blackmagic-debug/blackmagic/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds for hardware native (`make PROBE_HOST=native`)
* [x] It builds as BMDA (`make PROBE_HOST=hosted`)
* [x] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do

### CODE REVIEW

File: src/target/adiv5.c
Comment (dragonmux) [inline src/target/adiv5.c:99]: Do we know from the spec or the architecture manuals etc what these "R" entries should be? Or even, should they be valid strings rather than NULL for now to make it more obvious when the code hits one of these unknown/reserved CID values? The comment talks about saving space but having the table store null pointers would save more and be more obvious if we see one in the wild.

**Review by dragonmux — CHANGES_REQUESTED** [review]
This is looking good. There's one item that we have queries on that this touches but with that resolved, think this is good for merge.

Might we suggest, given this touches the Cortex-A and Cortex-M core support, that you give feature/arm-cortex-refactoring a look as that changes a fair whack to do with the Cortex code, as does feature/cortex-r-support which is built on top of the previously mentioned branch

File: src/target/adiv5.c
Comment (perigoso) [reply src/target/adiv5.c]: In looking into this I refactored `adiv5_ap_type`, which seemed to have a mistake, the function could look up from indexes 0 to 7, but only 6 entries existed

Comment (dragonmux) [reply src/target/adiv5.c]: The structure of adiv5_ap_type() was built around this table from the ADIv5 spec:
image

This is why the table and function was laid out how it was.

Comment (perigoso) [reply src/target/adiv5.c]: Ah! I looked for that table but it's not in the ADIv5.2 spec I was looking at, or at least not under that name (or maybe I just missed it)

But there was in fact a problem (not the missing `> 0` check), the table was doing a simple lookup for `type -1` but the type `0x3` does not exist so the table was missing one empty entry, every type after `0x3` would have been showing the incorrect string, and type `8` would be out of bounds

I corrected the values in my commit

Comment (dragonmux) [reply src/target/adiv5.c]: Oh, great catch! that apparently completely blind-sided us. We'll re-review with that in mind

File: src/target/adiv5.c
Comment (dragonmux) [inline src/target/adiv5.c:282]: This is looking much better. Given the `ap_class` is supposed to be 8 per the table for this first block of `ap_type`'s, it is probably a good idea to pivot this slightly to first checking `ap_class` then switching on `ap_type` (and yes, this was a flaw in our original logic too), but we'd be happy merging this as-is if you don't want to go through that to make this exactingly correct.

**Review by dragonmux — COMMENTED** [review]
Only have one comment that, once answered, leaves us happy to merge this. Great work! We look forward to finding a resolution to the `enum` problem too in a follow-up PR

File: src/target/adiv5.c
Comment (perigoso) [reply src/target/adiv5.c]: Good catch, added the additional validation

**Review by dragonmux — APPROVED** [review]
LGTM, we'll merge this once we've given it a test against some local parts to make sure things seem sane.

**Comment by dragonmux** [comment]
Looks good, seems to work well. Merging!

**Pull request merged**
